pkgs/checks/lib/src/checks.dart

```diff
@@ -181,16 +181,22 @@ abstract class Context<T> {
   Future<void> expectAsync<R>(Iterable<String> Function() clause,
       FutureOr<Rejection?> Function(T) predicate);
 
-  /// Report that this check may fail asynchronously at any point in the future.
+  /// Expect that [predicate] will not invoke the passed callback with a
```

Comment:
If we are expecting that it will _not_ ever invoke it with a rejection, then shouldn't the name be `expectNoAsyncFailure` or some such?

Reply:
`expectNoAsyncFailure` would work for me.

I also just thought of `expectUnawaited` which might give a better indication of why it would be used.

Reply:
`expectUnawaited` might also convey that the only "useful" side effect is the error which will always be uncaught async.

Reply:
Actually `expectUnawaited` sounds pretty reasonable to me 👍